src/test/java/com/internship/backend/service/ReservationServiceTest.java

```diff
@@ -1,12 +1,168 @@
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.InjectMocks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
 
+@ExtendWith(MockitoExtension.class)
 class ReservationServiceTest {
 
+    @Mock
+    private ReservationRepository reservationRepository;
+
+    @Mock
+    private TennisCourtRepository tennisCourtRepository;
+
+    @Mock
+    private PriceRepository priceRepository;
+
+    @InjectMocks
+    private ReservationService reservationService;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void addReservation_validReservation() throws InvalidDateException, ReservationAlreadyExists {
+        //arrange
+        Reservation reservation = new Reservation();
+        NewDate startTime = new NewDate(0, 17, 18, 7, 2024);
+        NewDate endTime = new NewDate(0, 19, 18, 7, 2024);
+        reservation.setStartTime(startTime);
+        reservation.setEndTime(endTime);
+
+        when(reservationRepository.findAll()).thenReturn(new ArrayList<>());
+        when(reservationRepository.save(reservation)).thenReturn(reservation);
+
+        //act
+        Reservation result = reservationService.addReservation(reservation);
+
+        //assert
+        assertNotNull(result);
+        assertEquals(reservation, result);
+    }
+
+    @Test
+    void addReservation_invalidReservation() throws InvalidDateException, ReservationAlreadyExists {
+        //arrange
+        Reservation reservation = new Reservation();
+        NewDate startTime = new NewDate(0, 17, 18, 7, 2024);
+        NewDate endTime = new NewDate(0, 20, 18, 7, 2024);
+        reservation.setStartTime(startTime);
+        reservation.setEndTime(endTime);
+        reservation.setTennisCourt(new TennisCourt());
+
+        //act
+        InvalidDateException exception = assertThrows(InvalidDateException.class, () -> {
+            reservationService.addReservation(reservationService.addReservation(reservation));
+        });
+
+        //assert
+        assertEquals("The reservation duration must be exactly 2 hours", exception.getMessage());
+    }
+
+    @Test
+    void getAllReservations() {
+        //arrange
+        List<Reservation> reservations = new ArrayList<>();
+
+        when(reservationRepository.findAll()).thenReturn(reservations);
+
+        //act
+        List<Reservation> result = reservationService.getAllReservations();
+
+        //assert
+        assertEquals(result, reservations);
+        verify(reservationRepository, times(1)).findAll();
+    }
+
     @Test
-    void getAvailableTennisCourts() {
+    void update_Valid() throws ReservationAlreadyExists {
```

Comment:
you could rename the tests more suggestively. for example this test can be renamed shouldUpdateWhenReservationIsValid or something